File: billing-service/src/main/java/com/kleff/billingservice/datalayer/Invoice/Invoice.java
package com.kleff.billingservice.datalayer.Invoice;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.sql.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "invoices")
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String invoiceId;
    String projectId;
    Date startDate;
    Date endDate;
    InvoiceStatus status;
    BigDecimal totalCPU;
    BigDecimal totalRAM;
    BigDecimal totalSTORAGE;
    BigDecimal subtotal;
    BigDecimal taxes;
    BigDecimal total;
    BigDecimal totalPaid;
}

File: frontend/src/features/billing/types/Invoice.ts
export type Invoice = {
  invoiceId: string;
  projectId: string;
  startDate: Date;
  endDate: Date;
  status: string;
  totalCPU: number;
  totalRAM: number;
  totalSTORAGE: number;
  subtotal: number;
  taxes: number;
  total: number;
  totalPaid: number;
};
